src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -50,17 +55,15 @@ public partial struct SyntaxValueProvider
         // changed. CreateSyntaxProvider will have to rerun all incremental nodes since it passes along the
         // SemanticModel, and that model is updated whenever any tree changes (since it is tied to the compilation).
         var syntaxTreesProvider = _context.CompilationProvider
-            .SelectMany(static (c, _) => c.SyntaxTrees)
+            .SelectMany((compilation, cancellationToken) => compilation.SyntaxTrees
+                .Select(tree => GetTreeInfo(tree, syntaxHelper, cancellationToken))
+                .Where(info => info.ContainsGlobalAliases || info.ContainsAttributeList))
```

Comment:
this is the critical part.  the compilation passes us its list of syntax trees, and we immediately filter down to those of interest.  We must do this first, prior to passing out the `SyntaxTree[]` to avoid the massive allocations in that output section of this incremental node.  Compilations can have thousands of trees, so this greatly filters down what actual trees we pass along (as well as all the processing of those output nodes in later stages of the incremental pipeline).this is the critical part.  the compilation passes us its list of syntax trees, and we immediately filter down to those of interest.  We must do this first, prior to passing out the `SyntaxTree[]` to avoid the massive allocations in that output section of this incremental node.  Compilations can have thousands of trees, so this greatly filters down what actual trees we pass along (as well as all the processing of those output nodes in later stages of the incremental pipeline).